# exemplos 3: números

In [2]:
#importar o comando pythontutor para gerar os links de cada exemplo para o site de visualização
from python_tutor_magic import pythontutor

# tipos de números

In [35]:
#%%pythontutor

#inteiros
#número sem ponto
i=1
#opcionalmente underline para separar milhares
j=10_000

#função type
print(type(i))

#float (representação de reais, truncada em aproximadamente 16 algarismos significativos)
#casa decimal separada por ponto
r=8.314
#pode notação científica (e maiusculo ou minusculo)
na=6.02e23
kb=1.346E-23
print(type(r))

#complexo
c=3.1+.5j
print(type(c))

#o j indica complexo se estiver justaposto a direita de um número

#complexo
print(1j)
#complexo
print(.1j)
#uma variável chamada j1
# print(j1)
#uma multiplicação com a variável j
print(2*j)
# também não pode espaço
# print(2 j)

i,j,r,na,kb,c

<class 'int'>
<class 'float'>
<class 'complex'>
1j
0.1j
20000


(1, 10000, 8.314, 6.02e+23, 1.346e-23, (3.1+0.5j))

# conversões

In [7]:
#%%pythontutor

print('qual o número de componentes')

resposta=input()
print(type(resposta))

# a função de conversão tem o nome do tipo
ncomp=int(resposta)
print(type(ncomp))

qual o número de componentes
2
<class 'str'>
<class 'int'>


# cálculos

In [32]:
#%%pythontutor

i=7
j=2

a=i+j
b=i-j
c=i*j
d=i/j
e=i**j

f=i%j
g=i//j

i,j,a,b,c,d,e,f,g

(7, 2, 9, 5, 14, 3.5, 49, 1, 3)

# conversões automáticas

In [31]:
#%%pythontutor

i=1
r=2.5
c=3.9+.2j

a=i+r
b=i+c
c=r+c

i,r,c,a,b,c

(1, 2.5, (6.4+0.2j), 3.5, (4.9+0.2j), (6.4+0.2j))

# funções da calculadora científica

In [59]:
#%%pythontutor
 
# comando import, carrega um módulo suplementar
import math as m
# a palavra chave as indica que vamos nos referir ao modlo math pelo nome m

#o ponto indica a que modulo pertencem as funções
a=m.sin(0)
b=m.cos(0)
c=m.exp(1)
d=m.log(10)
e=m.log10(10)
a,b,c,d,e

(0.0, 1.0, 2.718281828459045, 2.302585092994046, 1.0)

# algumas constantes

In [39]:
#%%pythontutor

import math as m
m.pi,m.e

(3.141592653589793, 2.718281828459045)

# precedencia
1. parênteses
1. **
1. * / % //
1. + -

In [50]:
#%%pythontutor

a=2**10+5*3/2/4
b=((2**10)+(((5*3)/2)/4))
c=2**((10+5)*(3/(2/4)))

d=2/3/4
e=(2/3)/4
f=2/(3*4)
g=2/(3/4)

a,b,c,d,e,f,g

(1025.875,
 1025.875,
 1.2379400392853803e+27,
 0.16666666666666666,
 0.16666666666666666,
 0.16666666666666666,
 2.6666666666666665)

# expressões com as variáveis

In [52]:
#%%pythontutor

x=3
print(x)

# = EM PROGRAMAÇÃO
# NÃO DEFINE UMA EQUAÇÃO ALGÉBRICA

x=2*x+1
# x <= 2*x+1
#      2*3+1
#      7
# x <= 7
print(x)

# Avaliar as operações do lado direito
# Vincular o resultado ao nome do lado esquerdo

3
7


# EXEMPLO

In [58]:
#%%pythontutor

energia=50
print(energia)

calor_cedido=10
energia=energia-calor_cedido
print(dinheiro)

trabalho_recebido=30
energia=energia+trabalho_recebido
print(energia)

50
10
70


# EXERCICIOS

1. calcular o perimetro e area de um quadrado de lado 1.3 m
1. calcular o perimetro e area de um círculo de raio 2.1 m
1. calcular quantos segundos tem em um ano
1. calcular o rendimento em um ano de uma aplicação com juros de .51% ao mês
1. calcular o volume de 1 mol de gas ideal nas CNTP